context/app/static/js/theme/theme.stories.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import { Box, Typography, Table, TableBody, TableCell, TableHead, TableRow } from '@mui/material';
+import theme from './theme';
+
+type PaletteColorProps = {
+  color: (typeof theme.palette)[keyof typeof theme.palette];
+  name: string;
+};
+
+function hex2rgb(c: string) {
```

Comment:
Did you pull these utils from somewhere?

Reply:
I based the `hex2rgb` solution on a StackOverflow answer which only showed the basic `parseInt` piece: https://stackoverflow.com/a/69353003

Since that logic only worked with 7-character hex colors (i.e. `#RRGGBB`) and I wanted to be sure that every edge case was covered, I extended it to be index-based in order to allow handling of short hex colors (`#rgb` like `#fff` white) and RGBA values (though the alpha parsing logic may need some minor revisions after looking at it with fresh eyes). I'll fix that piece of the logic and add crediting for the base to make sure all the tracks are covered.

The `rgb2hex` solution was easier to write since I could reliably pull out all the numeric digits from the original string and turn them into hexadecimals.